backend/backup_media.js

```diff
@@ -0,0 +1,171 @@
+require('dotenv-flow').config();
+const mysql = require('mysql2');
+const fs = require('fs');
+const crypto = require('crypto');
+const fetch = require('node-fetch');
+const sizeOf = require('image-size');
+const setTimeoutPromise = require('node:timers/promises').setTimeout;
+
+const MEDIA_TYPE_IMG = 'img';
+const MEDIA_TYPE_VIDEO = 'video';
+const SLEEP_BETWEEN_BATCHES_MS = 1000;
+const SLEEP_BETWEEN_MEDIA_ITEMS_MS = 100;
+const BATCH_SIZE = 1000;
+const MEDIA_DOWNLOAD_REQUEST_TIMEOUT_MS = 60000;
+const START_DOWNLOADING_FROM_DATE = '2022-01-01';
+
+const config = {
+  host: process.env.MYSQL_HOST || 'localhost',
+  port: parseInt(process.env.MYSQL_PORT) || 3306,
+  user: 'root',
+  password: process.env.MYSQL_ROOT_PASSWORD || 'orbitar',
+  database: process.env.MYSQL_DATABASE || 'orbitar_db',
+  dir: './backup-media',
+  dbFile: './backup-media/index.csv'
+};
+
+const sleep = ms => new Promise(r => setTimeout(r, ms));
+
+const download = async (url, dest, controller) => {
+    if (!url.match(/^(\/\/|http:|https:)/)) {
+      console.log(`Invalid URL: `, url);
+      return false;
+    }
+    if (url.match(/^\/\//)) {
+      url = 'https:' + url;
+    }
+    try {
+      const response = await fetch(url, { signal: controller.signal });
+      const responseBinary = await response.buffer();
+      let writeResult = await fs.promises.writeFile(dest, responseBinary);
+      await sleep(1000);
+      console.log(`Wrote ${responseBinary.length} bytes to ${dest} for ${url}`);
+      return controller.abort();
+    } catch (e) {
+      console.log(e);
+      return false;
+    }
+};
+
+const downloadRequestTimeout = async (ms, url, controller) => {
+  try {
+    await setTimeoutPromise(ms, undefined, { signal: controller.signal });
+    controller.abort();
+    console.log(`Request to fetch ${url} is aborted due to slow response`);
+    return false;
+  } catch (error) {
+    return false;
+  }
+};
+
+const connection = mysql.createConnection({
+  host: config.host,
+  user: config.user,
+  password: config.password,
+  database: config.database
+});
+
+try {
+  fs.mkdirSync(config.dir);
+} catch (e) {}
+if (!fs.existsSync(config.dbFile)) {
+  fs.openSync(config.dbFile, 'w+');
+  fs.appendFileSync(config.dbFile, 'item_type,item_id,media_type,media_src,media_id,img_width,img_height  ' + '\n');
+} else {
+  fs.openSync(config.dbFile, 'w+');
+}
```

Comment:
No need to open file when not using the returned handle.
```suggestion
}
```